doc/tutorial/concepts/index.md

```diff
@@ -0,0 +1,290 @@
+# C++ Concepts {#tutorial_concepts}
+
+[TOC]
+
+This tutorial introduces "C++ Concepts", a feature of C++20 (and available to some extent in older GCC versions).
+You will learn the terminology used in the context of concepts and how to use SeqAn3's concepts in your application.
+
+\tutorial_head{Moderate, 60 min, \ref setup\, \ref tutorial_alphabets,}
+
+This tutorial teaches the very basics of working with concepts. For more background and information on how to implement
+your own concepts, we recommend:
+  * A well-readable [paper](http://www.stroustrup.com/good_concepts.pdf) with motivation and historical background.
+  * The (rather dense) [documentation on cppreference](https://en.cppreference.com/w/cpp/language/constraints).
+
+
+# Constraints
+
+## Introduction
+
+In C++20 a typical pre-C++20 template is said to be *unconstrained*, i.e. it can be instantiated with any type:
+
+```cpp
+template <typename t>
+t add(t const v1, t const v2)
+{
+    return v1 + v2;
+}
+
+int main()
+{
+    return add(1, 3); // instantiates add<int>()
+}
+```
+
+Of course this will not actually compile for all types, because the function template **implicitly requires** that
+types provide a `+` operator.
```

Comment:
I would also add a small paragraph of how this improves readability of compiler errors when constraining the type with a concept.

Reply:
yep, good idead